### PR TITLE
Calculate sum of squared deviations for IntervalView

### PR DESCRIPTION
**Pull request by songy23**
Fix https://github.com/census-instrumentation/opencensus-java/issues/649

Calculate sum of squared deviations by using the Parallel algorithm. Thanks @isturdy for providing a solution to this issue.

### CODE REVIEW

File: core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java [hunk 1]
```diff
@@ -299,6 +299,18 @@ void combine(MutableAggregation other, double fraction) {
       checkArgument(this.bucketBoundaries.equals(mutableDistribution.bucketBoundaries),
           "Bucket boundaries should match.");
 
+      // Algorithm for calculating the combination of sum of squared deviations:
+      // https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm.
+      double delta = mutableDistribution.mean - this.mean;
+      double ma = this.sumOfSquaredDeviations * (this.count - 1);
+      double mb = mutableDistribution.sumOfSquaredDeviations * (mutableDistribution.count - 1);
```
Comment (isturdy) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java]: I think these should just be sumOfSquaredDeviations--the pseudocode from Wikipedia takes variances as arguments (using the unbiased estimator formula) and thus needs to multiply by N-1 to get back the sum of the squared deviations, but you already have the sum rather than the variance.

The test looks correct, but both the distributions have count == 2, so multiplying by count - 1 does nothing--I think it will fail if you give it a 3+ element distribution.

Comment (songy23) [reply core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java]: You're right, fixed. Also added a test for the case when there are >2 values.

File: core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java [hunk 2]
```diff
@@ -299,6 +299,16 @@ void combine(MutableAggregation other, double fraction) {
       checkArgument(this.bucketBoundaries.equals(mutableDistribution.bucketBoundaries),
           "Bucket boundaries should match.");
 
+      // Algorithm for calculating the combination of sum of squared deviations:
+      // https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm.
+      if (this.count + mutableDistribution.count > 0) {
+        double delta = mutableDistribution.mean - this.mean;
+        this.sumOfSquaredDeviations = this.sumOfSquaredDeviations
```
Comment (isturdy) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java]: Could be a bit more concise (and parallel with the other updates) with +=, but I think it obscures the correspondence to the linked algorithm--your call.

File: core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java [hunk 3]
```diff
@@ -299,6 +299,16 @@ void combine(MutableAggregation other, double fraction) {
...
+            + mutableDistribution.sumOfSquaredDeviations
+            + delta * delta * this.count * mutableDistribution.count
```
Comment (isturdy) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java]: I would prefer Math.pow(delta, 2).

File: core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java [hunk 4]
```diff
@@ -170,7 +170,7 @@ void record(TagContext tags, long value, Timestamp timestamp) {
     for (int i = 0; i < columns.size(); ++i) {
       TagKey tagKey = columns.get(i);
       if (!tags.containsKey(tagKey)) {
-        // replace not found key values by “unknown/not set”.
+        // replace not found tag values with null.
```
Comment (isturdy) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java]: Is this related to the other changes? If not I would prefer a separate pull. Regardless, this seems misleading--null suggests an empty string to me, not "UNKNOWN_TAG_VALUE".

File: core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java (hunk 3 above)
Comment (songy23) [reply core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java]: Updated.

File: core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java (hunk 2 above)
Comment (songy23) [reply core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java]: I'll prefer to leave this as it is - to be clear and more consistent with the formula.

File: core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java (hunk 4 above)
Comment (songy23) [reply core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java]: Sure, I'll make another PR to fix this.

**Comment by songy23** [comment]
Thanks!
